File: Samples/XObjects, watermarks and backgrounds/RemoveXObjectsWatermarks/VB.NET/RemoveXObjectsWatermarks.vb
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class RemoveXObjectsWatermarks
        Public Shared Sub Main()
            ' NOTE: 
            ' When used in trial mode, the library imposes some restrictions.
            ' Please visit http://bitmiracle.com/pdf-library/trial-restrictions
            ' for more information.

            Dim pathToFile As String = "RemoveXObjectsWatermarks.pdf"

            Using pdf As New PdfDocument("..\Sample Data\DocumentWithWatermark.pdf")
                For Each page As PdfPage In pdf.Pages
                    ' remove first XObject (in this case it's a watermark)
                    page.XObjects.RemoveAt(0)
                Next

                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace

File: Samples/XObjects, watermarks and backgrounds/RemoveXObjectsWatermarks/C#/RemoveXObjectsWatermarks.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class RemoveXObjectsWatermarks
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            string pathToFile = "RemoveXObjectsWatermarks.pdf";

            using (var pdf = new PdfDocument(@"..\Sample Data\DocumentWithWatermark.pdf"))
            {
                foreach (PdfPage page in pdf.Pages)
                {
                    // Remove the first XObject. Here, it's a watermark.
                    page.XObjects.RemoveAt(0);
                }

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }
    }
}
